### ISSUE TITLE
Icons in segments.

### ISSUE THREAD

**Issue by marcusadmortati** [Larkinabout/fvtt-challenge-tracker#14]
Could it be that in Challenger Tracker it would be possible in some future by icons in the segments as in the figure?
https://i.imgur.com/fV56t4i.png

**Comment by megahead11** [comment]
I'm assuming you mean you want to be able to segment an image? If that's the case, I wouldn't mind seconding this, especially if there was a foreground and background image option. 

I believe the clocks module originally had the ability to use custom images for the tracker, split up into sections.

**Comment by Larkinabout** [comment]
There's a few different ways I see that this could be interpreted and achieved:

- The option to add foreground and background images then just clip the foreground image where the segments are not filled. This wouldn't be adaptive to changes to the number of segments, so you'd need to change the images as needed, but would be the quickest to implement.
- A sort of extension to the above option would be to support groups of images (themes), with something in the filename to identify foreground/background and how many segments the image serves. Then we could pick the correct set of images whenever the number of segments changes.
- The option to add images then choose which segment number each image will sit on top of (or possibly between). This would support placing icons as individual elements. This is how I interpreted @marcusadmortati's original request.
- Like above, but allow each image to be dragged around on the challenge tracker. This is probably the most adaptable and wouldn't need to programatically handle changes to the number of segments as you'd just move the images yourself as needed.
- Do a mixture of the above in some form to support full foreground/background images as well as individual icons.

**Comment by marcusadmortati** [comment]
I mean assigning a small image to each segment. For each segment an image. The image I showed is a dungeon clock, what happens every hour, each segment has one turn: in the first the dm must check for random encounters, in the second if there are collapses and so on...

**Comment by marcusadmortati** [comment]
I mean putting one image or more than one in each segment. To indicate, for example:"now it's time to check if a monster appears."
https://i.imgur.com/k3zZefE.png

**Comment by megahead11** [comment]
Technically speaking, you could accomplish that by having the icons on the edges of the clock too, so by using a background image. Consequently, this also works for foreground images too, as the icon would be on the clock pieces.

**Comment by brunocalado** [comment]
The icon idea looks pretty bad.

I think, a better solution to achieve the same thing would be let the user choose two custom images. But, this would be extra work for the user who have to design the images. He would have to have pieces that match the clock size.

**Comment by megahead11** [comment]
Yeah, more or less what I suggested with the background/foreground images. It's easier to just have the icon in the clock picture itself. If you're using a custom clock image, I would imagine you had to size it correctly anyways :P

**Comment by Larkinabout** [comment]
This is partly added in release 1.0. I'll leave this issue open though as I'd still like to try and tackle placeable icons at some point.
